UnitystationLauncher/Models/ServerManager.cs

```diff
@@ -78,6 +84,25 @@ private async Task<IReadOnlyList<ServerViewModel>> GetServerList()
             return newServerList;
         }
 
+        private bool HasTrustedUrlSource(Server server)
+        {
+            const string trustedHost = "unitystationfile.b-cdn.net";
+            var urls = new[] { server.WinDownload, server.OsxDownload, server.LinuxDownload };
+            foreach (var url in urls)
+            {
+                if (string.IsNullOrEmpty(url))
+                {
+                    return false;
+                }
+                var uri = new Uri(url);
+                if (uri.Scheme != "https" && uri.Host != trustedHost)
+                {
+                    return false;
+                }
+            }
+            return true;
```

Comment:
Don't parse this yourself use https://docs.microsoft.com/en-us/dotnet/api/system.uri.trycreate?view=netframework-4.7.2#System_Uri_TryCreate_System_String_System_UriKind_System_Uri__ with `Absolute` `UriKind` instead.

Reply:
I have no clue how I misread the code and didn't see that you now use `Uri`.